File: 2000 98 & ME/QuickStart/aspplus/samples/webforms/busobjs/DataObj.cs

namespace DataLayer 
{
using System;
using System.Data;
using System.Data.SQL;

  public class DataObj 
  {
    private String _connStr;

    public DataObj() 
    {
      _connStr = null;
    }

    public DataObj(String connStr) 
    {
      _connStr = connStr;
    }

    public String ConnectionString 
    {
      get 
      {
        return _connStr;
      }
      set 
      {
        _connStr = value;
      }
    }

    public DataView GetCategories() 
    {
      SQLConnection myConnection = new SQLConnection(_connStr);
      SQLDataSetCommand myCommand = new SQLDataSetCommand("select distinct CategoryName from Categories", myConnection);

      DataSet ds = new DataSet();
      myCommand.FillDataSet(ds, "Categories");

      return ds.Tables["Categories"].DefaultView;
    }

    public DataView GetProductsForCategory(String category) 
    {
      SQLConnection myConnection = new SQLConnection(_connStr);
      SQLDataSetCommand myCommand = new SQLDataSetCommand("select ProductName, ImagePath, UnitPrice, c.CategoryId  from Products p, Categories c where c.CategoryName='" + category + "' and p.CategoryId = c.CategoryId", myConnection);

      DataSet ds = new DataSet();
      myCommand.FillDataSet(ds, "Products");

      return ds.Tables["Products"].DefaultView;
    }
  }

}

File: 2000 98 & ME/QuickStart/aspplus/samples/webforms/busobjs/VB/DataObj.vb
Imports System
Imports System.Data
Imports System.Data.SQL

Namespace DataLayer 

  Public Class DataObjVB

    Private _connStr As String

    Public Overloads Sub New() 

      MyBase.New()
      _connStr = ""
    End Sub

    Public Overloads Sub New(ConnStr As String) 

      MyBase.New()
      _connStr = ConnStr
    End Sub

    Public Property ConnectionString As String

      Get 
        Return _connStr
      End Get

      Set
        _connStr = Value
      End Set
    End Property

    Public Function GetCategories() As DataView

      Dim DS As DataSet
      Dim MyConnection As SQLConnection
      Dim MyCommand As SQLDataSetCommand

      MyConnection = New SQLConnection(_connStr)
      MyCommand = New SQLDataSetCommand("select distinct CategoryName from Categories", MyConnection)

      DS = new DataSet()
      MyCommand.FillDataSet(DS, "Categories")

      Return DS.Tables("Categories").DefaultView
    End Function

    Public Function GetProductsForCategory(Category As String) As DataView

      Dim DS As DataSet
      Dim MyConnection As SQLConnection
      Dim MyCommand As SQLDataSetCommand

      MyConnection = New SQLConnection(_connStr)

      MyCommand = New SQLDataSetCommand("select ProductName, ImagePath, UnitPrice, c.CategoryId  from Products p, Categories c where c.CategoryName='" & Category & "' and p.CategoryId = c.CategoryId", myConnection)

      DS = new DataSet()
      MyCommand.FillDataSet(DS, "Products")

      Return DS.Tables("Products").DefaultView
    End Function

  End Class

End Namespace